### PR TITLE
[tests]: O3-4092-Added tests for editing a clinical view

### PR DESCRIPTION
**Pull request by TrevorAntony**
## Requirements
- [x] This PR has a title that briefly describes the work done including a [conventional commit](https://o3-dev.docs.openmrs.org/#/getting_started/contributing?id=your-pr-title-should-indicate-the-type-of-change-it-is) type prefix and a Jira ticket number if applicable. See existing PR titles for inspiration.

#### For changes to apps
- [ ]  My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide) and [**design documentation**](https://zeroheight.com/23a080e38/p/880723-introduction).

#### If applicable
- [x] My work includes tests or is validated by existing tests.

## Summary
Added tests for editing a clinical view.
Additionally, consolidated repeated code into helper functions and moved common test data into constants (defaultExistingContent)

## Related Issue
https://openmrs.atlassian.net/browse/O3-4092

### CODE REVIEW

File: src/components/view-editor/view-editor.component.test.tsx [hunk 1]
```diff
@@ -15,40 +15,109 @@ beforeEach(() => {
...
+  });
+
+  it('should allow editing form configurations within menus', async () => {
+    const clinicalViewId = 'existing-clinical-view-id';
+    renderComponent(`/clinical-views-builder/edit/${clinicalViewId}`, clinicalViewId);
+    await verifyBasicElements();
+
+    const editTabDefinitionButton = screen.getByRole('button', { name: /renderChanges/i });
+    expect(editTabDefinitionButton).toBeInTheDocument();
+    await userEvent.click(editTabDefinitionButton);
```
Comment (CynthiaKamau) [inline src/components/view-editor/view-editor.component.test.tsx:122]: Is it possible to fire an event that edits something and validate that the edit happend ?

File: src/components/view-editor/view-editor.component.test.tsx [hunk 2]
```diff
@@ -15,40 +15,109 @@ beforeEach(() => {
...
   });
-});
+
+  it('should render the view editor with existing content', async () => {
+    const clinicalViewId = 'existing-clinical-view-id';
+    renderComponent(`/clinical-views-builder/edit/${clinicalViewId}`, clinicalViewId);
+    await verifyBasicElements();
+  });
+
+  it('should allow editing view content through the interactive builder', async () => {
```
Comment (CynthiaKamau) [inline src/components/view-editor/view-editor.component.test.tsx:92]: dtto

File: src/components/view-editor/view-editor.component.test.tsx [hunk 3]
```diff
@@ -15,40 +15,109 @@ beforeEach(() => {
...
+    await verifyBasicElements();
+
+    const editTabDefinitionButton = screen.getByRole('button', { name: /renderChanges/i });
+    expect(editTabDefinitionButton).toBeInTheDocument();
+    await userEvent.click(editTabDefinitionButton);
+  });
+
+  it('should properly update state when schema changes', async () => {
+    const clinicalViewId = 'existing-clinical-view-id';
+    renderComponent(`/clinical-views-builder/edit/${clinicalViewId}`, clinicalViewId);
```
Comment (CynthiaKamau) [inline src/components/view-editor/view-editor.component.test.tsx:130]: dtto, try adding a state and see if it changes

**Review by CynthiaKamau — COMMENTED** [review]
Thanks @TrevorAntony , left some comments

**Comment by TrevorAntony** [comment]
@CynthiaKamau  Made the suggested changes

**Review by CynthiaKamau — APPROVED** [review]

**Pull request merged**